docs/reference/projectyml.rst

```diff
@@ -100,72 +113,111 @@ example of a dev config:
+
+
+If the pip configuration is fully managed at the system level and secure, use ``use-system-config: true``
+and add any potential project-specific repo that is not managed at the system level through the ``extra-index-url``
+option e.g.:
 
 .. code-block:: yaml
 
     pip:
-        index-url: https://devpi.example.com/stable/
-        extra-index-url: []
+        extra-index-url: [https://my_repo.secure.company.com/repository/project-production]
         pre: false
         use-system-config: true
 
-.. note::
-    The options defined in the ``project.yml`` pip section will always take precedence over the corresponding pip options, even
-    when ``use-system-config`` is set to true (other pip-related environment variables are not overridden).
-
-    For example, in the production scenario above, if the following
-    pip environment variables were set by mistake on the server running the compiler: ``PIP_INDEX_URL=https://devpi.example.com/dev/``
-    and ``PIP_PRE=true``, the config used in the end would still be the one defined in the project.yml, namely
-    ``index-url=https://devpi.example.com/stable/`` and ``pre=false``.
 
-An alternative approach would be to configure all pip-related options through the system config. For example:
+On the other hand, if the pip configuration is not fully managed at the system level, make sure to disable ``use-system-config``
+and set ``index-url`` to the secure internal repo e.g.:
 
 .. code-block:: yaml
 
     pip:
-        index-url:
-        extra-index-url: []
-        pre:
-        use-system-config: true
+        index-url: https://my_repo.secure.company.com/repository/inmanta-production
+        extra-index-url: [https://my_repo.secure.company.com/repository/project-production]
+        pre: false
+        use-system-config: false
+
+.. note::
+    The ``pre`` and ``index-url`` options defined in the ``project.yml`` pip section will always take precedence over
+    the corresponding pip options, even when ``use-system-config`` is set to true.
 
-And set the following env variables:
+    The ``extra-index-url`` option will extend the extra indexes already set at the system level (if any).
 
-.. code-block:: bash
+    Other pip-related environment variables are not overridden.
 
-    export PIP_INDEX_URL=https://devpi.example.com/stable/
-    export PIP_PRE=false
+    To use the system's pip configuration without altering it, leave the corresponding option unset.
 
-In this scenario, pip options defined in env variables would be used over the system's pip config (if any).
 
 .. note::
 
-    Using netrc is the recommended way to set up authentication towards the index. See
-    this :ref:`section<setting_up_pip_index_authentication>` for more information.
+    Set up authentication towards the index using netrc. See this :ref:`section<setting_up_pip_index_authentication>`
+    for more information.
 
 .. _migrate_to_project_wide_pip_config:
 
 Migrate to project-wide pip config
```

Comment:
This section should mention the alternative of setting `use-system-config=true`.